murmur: Deprecate old name

felixsinger opened the pull request:
###### Motivation for this change

Based on 1.4.x branch.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

felixsinger committed:
mumble,murmur: 1.3.4 -> 1.4.230

Update version to 1.4.230

Signed-off-by: Felix Singer <felixsinger@posteo.net>
Signed-off-by: Franz Pletz <fpletz@fnordicwalking.de>

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -72,66 +49,49 @@ let
-    ] ++ lib.optional (!iceSupport) "CONFIG+=no-ice"
-      ++ lib.optional grpcSupport "CONFIG+=grpc";
+      "-D client=OFF"
+      "-D ice=OFF" # temporary off
```
todo

felixsinger replied:
CMake uses pkg-config and it can not find the zeroc-ice library. AFAICS the zeroc-ice package doesn't come with a pkg-config file. Needs more investigation.

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -72,66 +49,49 @@ let
+      "-D bundled-speex=OFF"
+      "-D bundle-qt-translations=OFF"
+      "-D update=OFF"
+      "-D overlay-xcompile=OFF"
```
todo

felixsinger replied:
won't enable this for now.

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -8,55 +8,32 @@
-      ++ (overrides.configureFlags or [ ]);
+    cmakeFlags = [
+      "-D g15=OFF"
+#     "-DRELEASE_ID=$(python ../scripts/mumble-version.py)"
```
todo

felixsinger replied:
RELEASE_ID parameter was dropped.

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -72,66 +49,49 @@ let
       repo = "mumble";
-      rev = version;
-      sha256 = "sha256-njovShQpW0JNeeE8VugdmGzDk3fKG8/fcJoBgdyTZug=";
+      rev = "69d9188e0c96996075cbf4555ebe050610ee63d0"; # latest commit of 1.4.x branch
```
todo: change to 1.4.0 stable release tag

felixsinger commented on pkgs/applications/networking/mumble/default.nix:
```diff
@@ -72,66 +50,52 @@ let
     nativeBuildInputs = [ qt5.qttools ];
-    buildInputs = [ libopus libsndfile speex qt5.qtsvg rnnoise ]
+    buildInputs = [ libopus libsndfile speex qt5.qtsvg rnnoise flac libogg libvorbis ]
       ++ lib.optional stdenv.isLinux alsa-lib
```
`meta.platforms` is set to `platforms.linux`. Do we need this check?

felixsinger replied:
Removed the check for `platforms.linux`. Instead, I let it depend on `!jackSupport`

K900 commented:
A stable release is out: https://www.mumble.info/blog/mumble-1.4.230/

felixsinger commented:
Thanks for pinging. Working on it!

K900 commented:
Happy to help test. Also, upstream just added a change that allows building with system rnnoise, might be worth looking into.

felixsinger committed:
murmur: Replace every occurrence of `murmur` with `mumble-server`

The name `murmur` is deprecated and shouldn't be used anymore, as
decided by the Mumble project. Thus, replace it with its successor
`mumble-server`.

Signed-off-by: Felix Singer <felixsinger@posteo.net>

mweinelt commented on nixos/modules/misc/ids.nix:
```diff
@@ -475,7 +475,7 @@ in
       vault = 115;
       #ripped = 116; # unused
-      murmur = 117;
+      mumble-server = 117;
```
Why do we need a persistent uid/gid at all? I believ we could be fine with DynamicUser. @hax404?

Infinisil replied:
See also NixOS/rfcs#52. Additionally I think just changing the name here would break things, because `/var/lib/nixos/{u,g}id-map` still maps `murmur` to 117. Not sure about the consequences.

K900 commented:
Any reason behind the rename? It'll break people's configs, and upstream seems to still use "murmur" all over the place.

felixsinger commented:
> Any reason behind the rename? It'll break people's configs, and upstream seems to still use "murmur" all over the place.

Please read the commit message.

felixsinger commented:
> Happy to help test. Also, upstream just added a change that allows building with system rnnoise, might be worth looking into.

Created a PR for porting it back to the 1.4.x branch. https://github.com/mumble-voip/mumble/pull/5464

K900 commented:
Adding an option alias helps, but it doesn't fully fix the issue. The database path should probably be conditional on `stateVersion` as well.

ldesgoui commented on nixos/modules/services/networking/mumble-server.nix:
```diff
@@ -3,18 +3,18 @@
-  configFile = pkgs.writeText "murmurd.ini" ''
-    database=/var/lib/murmur/murmur.sqlite
+  configFile = pkgs.writeText "mumble-server.ini" ''
+    database=/var/lib/mumble-server/mumble-server.sqlite
```
Won't that break existing installations?

felixsinger replied:
It does. I will let this depend on the `stateVersion`, as @K900 suggested. So people will have some time for migration.

fpletz commented:
Rebased. Fixed Pulseaudio & Pipewire support. Mumble is dynamically loading the libs.

arcnmx commented:
> Rebased. Fixed Pulseaudio & Pipewire support. Mumble is dynamically loading the libs.

[qt5.mkDerivation might already provide a wrapper you can use for this](https://github.com/arcnmx/nixexprs/blob/95d744c4c23416ab01e7909fb7f3746e83c725b9/pkgs/customized/default.nix#L309-L311) - would it be double-wrapped otherwise?

felixsinger commented:
> Rebased. Fixed Pulseaudio & Pipewire support. Mumble is dynamically loading the libs.

@fpletz Thanks!

mweinelt commented on nixos/modules/services/networking/mumble-server.nix:
```diff
@@ -3,18 +3,18 @@
-  configFile = pkgs.writeText "murmurd.ini" ''
-    database=/var/lib/murmur/murmur.sqlite
+  configFile = pkgs.writeText "mumble-server.ini" ''
+    database=/var/lib/mumble-server/mumble-server.sqlite
```
```suggestion
    database=${if lib.versionAtLeast config.system.stateVersion "22.05"
      then "/var/lib/mumble-server/mumble-server.sqlite"
      else "/var/lib/murmur/murmur.sqlite"}
```

Lassulus commented on pkgs/top-level/aliases.nix:
```diff
@@ -1182,7 +1182,7 @@ mapAliases ({
 
   mumble_git = pkgs.mumble; # added 2019-08-01
-  murmur_git = pkgs.murmur; # added 2019-08-01
+  mumble-server_git = pkgs.mumble-server; # added 2019-08-01
```
```suggestion
  murmur_git = pkgs.mumble-server; # added 2019-08-01
  murmur = pkgs.mumble-server; # added 2021-02-09
```

mweinelt replied:
Shouldn't we instead deprecrate the alias with a throw?

Lassulus replied:
also fine by me

infinisil reviewed (changes requested):
Just to highlight this again

felixsinger commented:
I created a new pull request for the version update. See #161663.

I will fix remaining issues in a few days, hopefully. At least the new version can be used then ;)

infinisil reviewed (commented):
Reminder for https://github.com/NixOS/nixpkgs/pull/161663#discussion_r816129395
